hack/validate-blocked-edges.py

```diff
@@ -1,9 +1,15 @@
 #!/usr/bin/env python
 
 import os
-
+import re
 import util
 
+# Risk names must be CamelCase to be assignable to condition.Reason
+# https://github.com/openshift/api/blob/8891815aa476232109dccf6c11b8611d209445d9/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1519-L1520C3
+# Note that our regex is stricter, we should never need names more complicated than this.
+NAME_RE_RAW = r'[A-Z]([A-Za-z0-9_]*[A-Za-z0-9])?$'
+NAME_RE = re.compile(NAME_RE_RAW)
```

Comment:
nit: you don't need the separate `NAME_RE_RAW`, because you can use the `pattern` method to get it back from the compiled regexp:

```console
$ python3 -c 'import re; NAME_RE = re.compile(r"[A-Z][A-Za-z0-9_]*$"); print("must match {!r}".format(NAME_RE.pattern))'
must match '[A-Z][A-Za-z0-9_]*$'
```